singlestoredb/fusion/handlers/job.py

```diff
@@ -0,0 +1,642 @@
+    from_start = FROM <integer>
+
+    # To end execution number
+    to_end = TO <integer>
+
+    Description
+    -----------
+    Shows the executions for the job with the specified ID within the specified range.
+
+    Arguments
+    ---------
+    * ``<job-id>``: The ID of the job to show executions for.
+    * ``<integer>``: The execution number to start from or end at.
+
+    Remarks
+    -------
+    * Use the ``FROM`` clause to specify the execution number to start from.
+    * Use the ``TO`` clause to specify the execution number to end at.
+    * To return more information about the executions, use the
+      ``EXTENDED`` clause.
+
+    Example
+    -------
+    The following command shows extended information on the executions for the job
+    with ID **job1**, from execution number **1** to **10**::
+
+        SHOW JOB EXECUTIONS 'job1'
+          FROM 1 TO 10
+          EXTENDED;
+    """
+
+    def run(self, params: Dict[str, Any]) -> Optional[FusionSQLResult]:
+        res = FusionSQLResult()
+        res.add_field('ExecutionID', result.STRING)
+        res.add_field('ExecutionNumber', result.INTEGER)
+        res.add_field('JobID', result.STRING)
+        res.add_field('Status', result.STRING)
+        res.add_field('ScheduledStartTime', result.DATETIME)
+        res.add_field('StartedAt', result.DATETIME)
+        res.add_field('FinishedAt', result.DATETIME)
+
+        jobs_manager = get_workspace_manager().organizations.current.jobs
+
+        executionsData = jobs_manager.get_executions(
+            params['job_id'],
+            params['from_start'],
+            params['to_end'],
+        )
+
+        if params['extended']:
+            res.add_field('SnapshotNotebookPath', result.STRING)
+
+            def fields(execution: Any) -> Any:
+                return (
+                    execution.execution_id,
+                    execution.execution_number,
+                    execution.job_id,
+                    execution.status.value,
+                    dt_isoformat(execution.scheduled_start_time),
+                    dt_isoformat(execution.started_at),
+                    dt_isoformat(execution.finished_at),
+                    execution.snapshot_notebook_path,
+                )
+        else:
+            def fields(execution: Any) -> Any:
+                return (
+                    execution.execution_id,
+                    execution.execution_number,
+                    execution.job_id,
+                    execution.status.value,
+                    dt_isoformat(execution.scheduled_start_time),
+                    dt_isoformat(execution.started_at),
+                    dt_isoformat(execution.finished_at),
+                )
+
+        res.set_rows([fields(execution) for execution in executionsData.executions])
+
+        return res
+
+
+ShowJobExecutionsHandler.register(overwrite=True)
+
+
+class ShowJobParametersHandler(SQLHandler):
+    """
+    SHOW JOB PARAMETERS job_id;
```

Comment:
What do you think about adding a `FOR` before the job ID here to make it read more like a sentence?

Reply:
Sounds good, I will change this and apply the same logic to `SHOW JOB EXECUTIONS job_id`